kasa/deviceconfig.py

```diff
@@ -1,10 +1,27 @@
-"""Module for holding connection parameters.
+"""Configuration for connecting directly to a device without discovery.
+
+If you are connecting to a newer KASA or TAPO device you can get the device
+via discovery or connect directly with :class:`DeviceConfig`.
+
+Example:
+    Discovery returns a list of discovered devices:
+
+    >>> from kasa import Discover, Credentials
+    >>> device = await Discover.discover_single(
+    >>>     "127.0.0.1",
+    >>>     credentials=Credentials("myusername", "mypassword"),
```

Comment:
Would you prefer getting #936 in before this PR, or should we just adjust these later on?